### DISCUSSION TITLE
elongated grains along arbitrary direction (not x, y, z direction) with Raster tessellation?

### DISCUSSION THREAD

**Discussion by damulaoshi** [neperfepx/neper#583] [Q&A] [answered]
Dear Romain,
Is there anyway to generate elongated grains along the arbitrary direction (not x, y, z directions) if I want to use raster tessellation?
For Voronoi tessellation, it works fine with the code:
neper -T -n from_morpho -domain "cube(300,300,300):rotate(0,0,1,25):scale(0.5,2,1)" -morpho "diameq:lognormal(30,3)" -transform "scale(2,0.5,1),rotate(0,0,1,-25)" -format tess -statcell vol,diameq,coo -morphooptistop "val=1e-2" -o elongated_test;

And the figure is here:
![image](https://user-images.githubusercontent.com/75380898/196375250-7be0ff74-9df3-48e0-8363-78a0d4108868.png)

But whenever I use Raster tessellation (the code below):
neper -T -n from_morpho -domain "cube(300,300,300):rotate(0,0,1,25):scale(0.5,2,1)" -morpho "diameq:lognormal(30,3)" -transform "scale(2,0.5,1),rotate(0,0,1,-25)" -tesrsize 100 -format tesr -statcell vol,diameq,coo -morphooptistop "val=1e-2" -statvox cell,coo -o elongated_raster;

Then the plot figure is very strange:
![image](https://user-images.githubusercontent.com/75380898/196375523-903ce2e6-a2f6-48ce-a1d8-4359d1679b24.png)

How to generate these elongated grains along arbitrary directions with Raster tessellation?

Thanks before!
Best regards

**Comment by rquey** [comment, marked as answer]
This looks like a nasty bug... but please provide a minimal working example as described [here](https://github.com/neperfepx/neper/issues/new?assignees=&labels=&template=bug_report.md&title=):

> Provide a *minimal working example* that can be used to reproduce the bug, that is, *complete* but *excluding unnecessary options*, *large inputs* and *excessive computation times*.  The example should include the program commands and other necessary input, and the terminal output.
